inference-engine/src/auto_plugin/auto_plugin.cpp

```diff
@@ -12,32 +12,51 @@
 #include <ie_core.hpp>
 #include <threading/ie_executor_manager.hpp>
 #include <ie_algorithm.hpp>
+#include <ngraph/opsets/opset1.hpp>
 
+#include <auto_plugin/auto_config.hpp>
 #include "auto_plugin.hpp"
+#include "ngraph_ops/convolution_ie.hpp"
+#include "ngraph_ops/deconvolution_ie.hpp"
 
 namespace AutoPlugin {
 namespace {
-    ConfigType mergeConfigs(ConfigType config, const ConfigType& local) {
-        for (auto && kvp : local) {
-            config[kvp.first] = kvp.second;
-        }
-        return config;
-    }
-
-    DeviceInformation SelectDevice(const std::vector<DeviceInformation>& metaDevices) {
-        for (auto& item : metaDevices) {
-            if (item.deviceName.find("CPU") == 0) {
-              return item;
+    std::string GetNetworkPrecision(const InferenceEngine::CNNNetwork &network) {
+        auto nGraphFunc = network.getFunction();
+        for (auto & node : nGraphFunc->get_ordered_ops()) {
+            if (std::dynamic_pointer_cast<ngraph::op::FakeQuantize>(node)) {
+                return METRIC_VALUE(INT8);
+            }
+            if (std::dynamic_pointer_cast<ngraph::opset1::Convolution>(node) ||
+                std::dynamic_pointer_cast<ngraph::opset1::GroupConvolution>(node) ||
+                std::dynamic_pointer_cast<ngraph::opset1::GroupConvolutionBackpropData>(node) ||
+                std::dynamic_pointer_cast<ngraph::opset1::ConvolutionBackpropData>(node) ||
+                std::dynamic_pointer_cast<ngraph::op::ConvolutionIE>(node) ||
+                std::dynamic_pointer_cast<ngraph::op::DeconvolutionIE>(node)) {
+                auto layerType = node->input(0).get_element_type().get_type_name();
```

Comment:
as far as I can see from [docs](https://docs.openvinotoolkit.org/latest/openvino_docs_ops_convolution_Convolution_1.html), convolution weights is a second argument.

Reply:
> as far as I can see from [docs](https://docs.openvinotoolkit.org/latest/openvino_docs_ops_convolution_Convolution_1.html), convolution weights is a second argument.

OK, @coneypo please check doc and double confirm in source code.

Reply:
I think you're right, just checked the doc opset1~opset6, and source code example `ngraph_api_examples`, it should be node->input(1). Thanks!